addons/account/tests/test_sequence_mixin.py

```diff
@@ -234,75 +243,61 @@ def test_sequence_draft_first_of_period(self):
         | 4    | `A`  | Cancel      | 2023-02-01 | `2023/02/0001` | -> Assert
         """
         move_a = self.test_move.copy({'date': '2023-02-01'})
-        self.assertEqual(move_a.name, 'MISC/22-23/02/0001')
+        self.assertMoveName(move_a, 'MISC/22-23/02/0001')
 
         move_b = self.test_move.copy({'date': '2023-02-02'})
-        self.assertEqual(move_b.name, '/')
+        self.assertMoveName(move_b, 'MISC/22-23/02/0001')
 
         move_b.action_post()
-        self.assertEqual(move_b.name, 'MISC/22-23/02/0002')
+        self.assertMoveName(move_b, 'MISC/22-23/02/0001')
 
+        # The first sequence slot is now taken by move_b, move_a's name and placeholder should be False.
         move_a.button_cancel()
-        self.assertEqual(move_a.name, 'MISC/22-23/02/0001')
+        self.assertMoveName(move_a, False)
 
     def test_journal_sequence(self):
-        self.assertEqual(self.test_move.name, 'MISC/15-16/01/0001')
+        self.assertMoveName(self.test_move, 'MISC/15-16/01/0001')
         self.test_move.action_post()
-        self.assertEqual(self.test_move.name, 'MISC/15-16/01/0001')
+        self.assertMoveName(self.test_move, 'MISC/15-16/01/0001')
 
         copy1 = self.create_move(date=self.test_move.date)
-        self.assertEqual(copy1.name, '/')
+        self.assertMoveName(copy1, False)
         copy1.action_post()
-        self.assertEqual(copy1.name, 'MISC/15-16/01/0002')
+        self.assertMoveName(copy1, 'MISC/15-16/01/0002')
 
         copy2 = self.create_move(date=self.test_move.date)
         new_journal = self.test_move.journal_id.copy()
         new_journal.code = "MISC2"
         copy2.journal_id = new_journal
-        self.assertEqual(copy2.name, 'MISC2/15-16/01/0001')
-
-        with Form(copy2) as move_form:  # It is editable in the form
-            with self.assertLogs('odoo.tests.form') as cm:
-                move_form.name = 'MyMISC/2016/0001'
-            self.assertTrue(cm.output[0].startswith('WARNING:odoo.tests.form.onchange:'))
-            self.assertIn('The sequence will restart at 1 at the start of every year', cm.output[0])
```

Comment:
Why are you removing this?

Reply:
Well these assertions were made on logs but the message of the logs was never shown to the user. With the new modifications, the message isn't even logged in the case of the test so either that or a different test imo. Here's why in details: 

Before this change, all moves would have a name so in `_onchange_name_warning`, `self._origin.name` would have a value resulting in entering the section of the method that decides which error message is to be show. However, as the move is the first of the journal, it's `highest_name` is falsy and `self.show_name_warning` will always be falsy as a consequence. This means that warning log exists but the message is not shown.

Now, as no name is ever assigned before posting the move, `self._origin.name` is falsy before first posting meaning that the message will still not be shown but it will also not even be logged.

If we don't want to remove this form test, I believe it would require to post the move, create a second one and test the behavior on that one.

Reply:
> were made on logs but the message of the logs was never shown to the user.

That's only in the form emulator for the tests. In practice, a warning is shown to the user.

> Now, as no name is ever assigned before posting the move

Then post it, reset to draft and change the value.
Or create a second one like suggested.